tests/unit_tests/simtel/test_simtel_io_event_reader.py

```diff
@@ -1,190 +1,198 @@
+    file_info_table["nsb_level"] = [1.0]
+
+    return shower_table, trigger_table, file_info_table
 
 
 @pytest.fixture
-def mock_hdf5_file():
-    with patch("tables.open_file") as mock_open:
-        mock_file = Mock()
-
-        # Mock reduced data
-        mock_reduced = Mock()
-        mock_reduced.nrows = 2
-        mock_reduced.col.side_effect = lambda x: {
-            "simulated_energy": np.array([1.0, 2.0]),
-            "x_core": np.array([100.0, 200.0]),
-            "y_core": np.array([150.0, 250.0]),
-            "shower_azimuth": np.array([0.1, 0.2]),
-            "shower_altitude": np.array([1.0, 1.1]),
-            "shower_id": np.array([1, 2]),
-            "area_weight": np.array([1.0, 1.0]),
-        }[x]
-        # Mock triggered data
-        mock_triggered = Mock()
-        mock_triggered.nrows = 2
-        mock_triggered.col.side_effect = lambda x: {
-            "triggered_id": np.array([0, 1]),
-            "array_altitudes": np.array([1.1, 1.2]),
-            "array_azimuths": np.array([0.2, 0.3]),
-            "telescope_list_index": np.array([0, 1]),
-        }[x]
-
-        mock_file.root.data.reduced_data = mock_reduced
-        mock_file.root.data.triggered_data = mock_triggered
-        mock_file.root.data.trigger_telescope_list_list = Mock()
-        mock_file.root.data.trigger_telescope_list_list.__len__ = lambda x: 2
-        mock_file.root.data.trigger_telescope_list_list.nrows = 2
-        mock_file.root.data.trigger_telescope_list_list.__getitem__ = (
-            lambda x, i: np.array([1, 2]) if i == 0 else np.array([2, 3])
-        )
-        mock_open.return_value.__enter__.return_value = mock_file
-
-        yield mock_open
-
-
-def test_init(mock_hdf5_file):
-    mock_file = mock_hdf5_file.return_value.__enter__.return_value
-    reader = SimtelIOEventDataReader(mock_file)
+def mock_fits_file(mock_tables, tmp_path):
+    """Create a mock FITS file with test data."""
+    test_file = tmp_path / "test.fits"
+    shower_table, trigger_table, file_info_table = mock_tables
+
+    shower_table.write(test_file, format="fits", overwrite=True)
+    trigger_table.write(test_file, format="fits", append=True)
+    file_info_table.write(test_file, format="fits", append=True)
+
+    return str(test_file)
+
+
+def test_reader_initialization(mock_fits_file):
+    """Test basic reader initialization."""
+    reader = SimtelIOEventDataReader(mock_fits_file)
     assert isinstance(reader.shower_data, ShowerEventData)
-    assert isinstance(reader.triggered_shower_data, ShowerEventData)
     assert isinstance(reader.triggered_data, TriggeredEventData)
+    assert len(reader.shower_data.simulated_energy) == 2
+    assert len(reader.triggered_data.telescope_list) == 2
 
 
-def test_read_event_data(mock_hdf5_file):
-    reader = SimtelIOEventDataReader(None)
-    shower, triggered_shower, triggered = reader.read_event_data(mock_hdf5_file)
+def test_telescope_filtering(mock_fits_file):
```

Comment:
Maybe add the case here for several telescopes in the list.

Reply:
The telescope list code will be changed soon for the reading of layouts from the DB. I would prefer not to touch the code now, as it will be changed in the next days anyway. OK for you?